build-tools/automation/yaml-templates/setup-test-environment.yaml

```
@@ -13,9 +13,14 @@ parameters:
   repositoryAlias: 'self'
   commit: ''
   xaprepareScenario: AndroidTestDependencies        # Use 'EmulatorTestDependencies' for agents that need the emulator installed
+  prepareSteps: []
 
 steps:
 
+- ${{ each step in parameters.prepareSteps }}:
+  - ${{ each pair in step }}:
+      ${{ pair.key }}: ${{ pair.value }}
+
```

You should be able to simplify this block with one line:

```suggestion
- ${{ parameters.prepareSteps }}
```